### PR TITLE
Sync cols for `pg_catalog` tables

### PR DESCRIPTION
**Pull request by matriv**
Please see individual commits.

Closes: #13418
Follows: #13420

### CODE REVIEW

File: server/src/main/java/io/crate/metadata/pgcatalog/PgTypeTable.java [hunk 1]
```diff
@@ -44,29 +45,41 @@ public static SystemTable<PGType<?>> create() {
...
+            .add("typsubscript", REGPROC, c -> REGPROC_ZERO)
+            .add("typelem", INTEGER, PGType::typElem)
             .add("typarray", INTEGER, PGType::typArray)
             .add("typinput", REGPROC, PGType::typInput)
             .add("typoutput", REGPROC, PGType::typOutput)
             .add("typreceive", REGPROC, PGType::typReceive)
             .add("typsend", REGPROC, PGType::typSend)
+            .add("typmodin", REGPROC, c -> REGPROC_ZERO)
+            .add("typmodout", REGPROC, c -> REGPROC_ZERO)
+            .add("typanalyze", REGPROC, c -> REGPROC_ZERO)
```
Comment (matriv) [inline server/src/main/java/io/crate/metadata/pgcatalog/PgTypeTable.java:68]: I think we want to return 0 here, as from psql I see the following for this col and it seems it has to do with range types:
```
                typname                 |      typanalyze       
----------------------------------------+-----------------------
 tsvector                               | ts_typanalyze
 int4range                              | range_typanalyze
 numrange                               | range_typanalyze
 tsrange                                | range_typanalyze
 tstzrange                              | range_typanalyze
 daterange                              | range_typanalyze
 int8range                              | range_typanalyze
 int4multirange                         | multirange_typanalyze
 nummultirange                          | multirange_typanalyze
 tsmultirange                           | multirange_typanalyze
 tstzmultirange                         | multirange_typanalyze
 datemultirange                         | multirange_typanalyze
 int8multirange                         | multirange_typanalyze
```

File: server/src/main/java/io/crate/metadata/pgcatalog/PgTypeTable.java [hunk 2]
```diff
@@ -44,29 +45,41 @@ public static SystemTable<PGType<?>> create() {
...
-            .add("typnamespace", INTEGER, c -> TYPE_NAMESPACE_OID)
+            .add("typsubscript", REGPROC, c -> REGPROC_ZERO)
+            .add("typelem", INTEGER, PGType::typElem)
             .add("typarray", INTEGER, PGType::typArray)
             .add("typinput", REGPROC, PGType::typInput)
             .add("typoutput", REGPROC, PGType::typOutput)
             .add("typreceive", REGPROC, PGType::typReceive)
             .add("typsend", REGPROC, PGType::typSend)
+            .add("typmodin", REGPROC, c -> REGPROC_ZERO)
+            .add("typmodout", REGPROC, c -> REGPROC_ZERO)
```
Comment (matriv) [inline server/src/main/java/io/crate/metadata/pgcatalog/PgTypeTable.java:67]: Not sure if this is correct, or we want to do something similar to https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/protocols/postgres/types/PGType.java#L133. Cockroach returns 0 here: https://github.com/cockroachdb/cockroach/blob/2e2e5c138e43c79d1d723e72326da05b6c5a1d84/pkg/sql/logictest/testdata/logic_test/pg_catalog#L1978
psql:
```
matriv=> select typname, typmodin from pg_type where typmodin != '0';
   typname    |      typmodin       
--------------+---------------------
 bpchar       | bpchartypmodin
 varchar      | varchartypmodin
 time         | timetypmodin
 timestamp    | timestamptypmodin
 timestamptz  | timestamptztypmodin
 interval     | intervaltypmodin
 timetz       | timetztypmodin
 bit          | bittypmodin
 varbit       | varbittypmodin
 numeric      | numerictypmodin
 _bpchar      | bpchartypmodin
 _varchar     | varchartypmodin
 _time        | timetypmodin
 _timestamp   | timestamptypmodin
 _timestamptz | timestamptztypmodin
 _interval    | intervaltypmodin
 _timetz      | timetztypmodin
 _bit         | bittypmodin
 _varbit      | varbittypmodin
 _numeric     | numerictypmodin
(20 rows)
```
```
matriv=> select typname, typmodout from pg_type where typmodout != '0';
   typname    |      typmodout       
--------------+----------------------
 bpchar       | bpchartypmodout
 varchar      | varchartypmodout
 time         | timetypmodout
 timestamp    | timestamptypmodout
 timestamptz  | timestamptztypmodout
 interval     | intervaltypmodout
 timetz       | timetztypmodout
 bit          | bittypmodout
 varbit       | varbittypmodout
 numeric      | numerictypmodout
 _bpchar      | bpchartypmodout
 _varchar     | varchartypmodout
 _time        | timetypmodout
 _timestamp   | timestamptypmodout
 _timestamptz | timestamptztypmodout
 _interval    | intervaltypmodout
 _timetz      | timetztypmodout
 _bit         | bittypmodout
 _varbit      | varbittypmodout
 _numeric     | numerictypmodout
(20 rows)
```

**Comment by matriv** [comment]
@mfussenegger

- I could squash commits after review (all the cols additions/removals) into 1, wdyt?
- I could "squash" the changelog entries about those additions/removals into 1, imho it looks a bit cleaner like this, and during release we can create a special section with all those points.

**Review by mfussenegger — APPROVED** [review]
> I could squash commits after review (all the cols additions/removals) into 1, wdyt?

Would also be fine to keep them.  No strong opinion

>  I could "squash" the changelog entries about those additions/removals into 1, imho though it looks a bit cleaner like this, and during release we can create a special section with all those points, or at least we can postpone this decision during the release?
 
Either way seems fine

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
@mergifyio rebase

**Comment by matriv** [comment]
retest this please

**Pull request merged**